src/audio/selector/selector_generic.c

```diff
@@ -121,17 +124,22 @@ static void sel_s32le_nch(struct comp_dev *dev, struct comp_buffer *sink,
 		}
 	}
 }
+#endif /* CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE */
 
 const struct comp_func_map func_table[] = {
+#if CONFIG_FORMAT_S16LE
 	{SOF_IPC_FRAME_S16_LE, 1, sel_s16le_1ch},
+	{SOF_IPC_FRAME_S16_LE, 2, sel_s16le_nch},
+	{SOF_IPC_FRAME_S16_LE, 4, sel_s16le_nch},
+#endif /* CONFIG_FORMAT_S16LE */
+#if CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE
 	{SOF_IPC_FRAME_S24_4LE, 1, sel_s32le_1ch},
```

Comment:
why do you treat S24/S32 together?

Reply:
I get at use sel_s32le_1ch, but it was mistake, thanks